syntax = "proto3";

import "device.proto";
import "u727.proto";
package mgw;

message MgwMsg {
    oneof Message {
        // messages for mgw <--> device communicate
        device.CommonRsp           response       = 20;
        device.SessionReq          sessionReq     = 21;
        device.SessionRsp          sessionRsp     = 22;
        device.StartOutputStream   startOutput    = 23;
        device.StopOutputStream    stopOutput     = 24;
        device.OutputStreamStatus  outputSta      = 25;
        device.StopPushingStream   stopPushing    = 26;
        device.SyncStatusReq       syncReq        = 27;
        device.SyncStatusRsp       syncRsp        = 28;
        device.ServerSessionReq    serSessionReq  = 29;
        device.ServerSessionRsp    serSessionRsp  = 30;
        device.QuerySource         querySource    = 31;
        device.SetPullAttr         setPullAttr    = 32;
        device.PushStreamReq       pushStreamReq  = 33;
        device.ServerKeepAlive     keepAlive      = 34;

        // messages for mgw <--> u727 communicate
        u727.SetDevBlacklist setBlackList = 1000;         // U727 -> MGW
        u727.SetLocalSvcPortReq setSvcPortReq = 1001;     // U727 -> MGW
        u727.SetLocalSvcPortReply setSvcPortReply = 1002; // MGW -> U727
        u727.GetLocalSvcPortReq getSvcPortReq = 1003;     // U727 -> MGW
        u727.GetLocalSvcPortReply getSvcPortReply = 1004; // MGW -> U727
        u727.StartStreamReq startStreamReq = 1005;        // U727 -> MGW
        u727.StartStreamReply startStreamReply = 1006;    // MGW -> U727
        u727.StopStreamReq  stopStream = 1007;            // U727 -> MGW
        u727.MgwDevOnlineNotify devOnline = 1008;         // MGW -> U727
        u727.MgwDevOfflineNotify devOffline = 1009;      // MGW -> U727
        u727.StreamStatusNotify streamStatus = 1010;     // MGW -> U727
        u727.DevStartStreamNotify devStartStreamNotify = 1011; // MGW -> U727
        u727.DevStopStreamNotify devStopStreamNotify = 1012;   // MGW -> U727
        u727.KeepAlive u727keepAlive = 1013; // U727 -> MGW -> U727
        u727.QueryOnlineDevReq queryOnlineDevReq = 1014; // U727 -> MGW
        u727.QueryOnlineDevReply queryOnlineDevReply = 1015; // MGW -> U727
        u727.QueryStreamStatusReq queryStreamReq = 1016; // U727 -> MGW
        u727.QueryStreamStatusReply queryStreamReply = 1017; // MGW -> U727
    }
}